src/si/angle.rs

```diff
@@ -0,0 +1,96 @@
+//! Angle (dimensionless).
+
+use super::angle_kind::AngleKind;
+
+quantity! {
+    /// Ratio (dimensionless).
+    quantity: Angle; "angle";
+    /// Ratio dimension (dimensionless).
+    dimension: ISQ<
+        Z0,     // length
+        Z0,     // mass
+        Z0,     // time
+        Z0,     // electric current
+        Z0,     // thermodynamic temperature
+        Z0,     // amount of substance
+        Z0>;    // luminous intensity
+    kind: AngleKind;
+    units {
+        /// SI derived unit of angle.  It is the angle subtended at the center of a circle by an
+        /// arc that is equal in length to the radius of the circle.
+        @radian: 1.0; "rad", "radian", "radians";
+        @revolution: 6.283_185_307_179_586_E0; "r", "revolution", "revolutions";
+        @degree: 1.745_329_251_994_329_5_E-2; "°", "degree", "degrees";
+        @gon: 1.570_796_326_794_896_7_E-2; "gon", "gon", "gons";
+        @mil: 9.817_477_E-4; "mil", "mil", "mils";
+        @minute: 2.908_882_086_657_216_E-4; "′", "minute", "minutes";
+        @second: 4.848_136_811_095_36_E-6; "″", "second", "seconds";
+    }
+}
+
+mod convert {
+    use super::*;
+
+    impl<U, V> ::lib::convert::From<V> for Angle<U, V>
+    where
+        U: ::si::Units<V> + ?Sized,
+        V: ::num::Num + ::Conversion<V>,
+    {
+        fn from(t: V) -> Self {
+            Angle {
+                dimension: ::lib::marker::PhantomData,
+                units: ::lib::marker::PhantomData,
+                value: t,
+            }
+        }
+    }
+
+    storage_types! {
+        use super::*;
+
+        impl<U> ::lib::convert::From<Angle<U, V>> for V
+        where
+            U: ::si::Units<V> + ?Sized,
+            V: ::num::Num + ::Conversion<V>,
+        {
+            fn from(t: Angle<U, V>) -> Self {
+                t.value
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    storage_types! {
+        use ::lib::f64::consts::PI;
+        use num::One;
+        use num_traits::FromPrimitive;
+        use si::angle as a;
+        use si::quantities::*;
+        use tests::Test;
+
+        #[test]
+        fn from() {
+            let r1: Angle<V> = Angle::<V>::from(V::one());
+            let r2: Angle<V> = V::one().into();
+            let _: V = V::from(r1);
+            let _: V = r2.into();
+        }
+
+        #[test]
+        fn check_units() {
+            Test::assert_eq(&Angle::new::<a::radian>(V::from_f64(2.0 * PI).unwrap()),
+                &Angle::new::<a::revolution>(V::one()));
+            Test::assert_eq(&Angle::new::<a::degree>(V::from_f64(360.0).unwrap()),
+                &Angle::new::<a::revolution>(V::one()));
+            // this test fails by ~1_E-15
```

Comment:
I'm not sure, I used 16 sigfigs of pi to compute it (because that's what sublime uses you when you eval `math.pi / 200.`), assert_approx_eq did the trick though, thanks for the tip!